### ISSUE TITLE
Change dynamic map in RUM methods to avoid lint error

### ISSUE THREAD

**Issue by scopendo** [DataDog/dd-sdk-flutter#143]
**Is your feature request related to a problem? Please describe.**
The signatures of logging methods, such as `DdLogs.info`, specify the optional `context` parameter as a `Map<String, Object?>`, whereas the signatures of RUM methods, such as `DdRum.addUserAction`, specify the optional `attributes` parameter as a `Map<String, dynamic>`.

Since many linting configurations seek to avoid "implicit dynamic", with such linting in place, this produces a Dart analysis message "error: Missing type arguments for map literal. (implicit_dynamic_map_literal at ...)".

```dart
DatadogSdk.instance.rum?.addUserAction(
  RumUserActionType.custom,
  'some_action',
  {}, // <== IDE shows syntax error highlighting here
);
```

**Describe the solution you'd like**
Whilst this can be addressed by explicitly typing the map, it would be nicer if the signatures of the RUM methods were changed to have the `attributes` parameter as `Map<String, Object?>`, which avoids the problem altogether and is consistent with the logging methods.

It might also make sense to rename the `context` parameter on logging methods to `attributes` for consistency, especially given that this is where they appear to end up in Datadog.  I appreciate that this doesn't affect our code since these are positional parameters.

**Comment by fuzzybinary** [comment]
@scopendo What linter / analysis rule are you using? I'd like to add it to the project in #148 to make sure we retain this level strictness moving forward (and catch anywhere else we might have missed)

**Comment by KarlChow92** [comment]
Thanks, was just looking for that as well. Seems like after enabling it we have more issues! 

![image](https://user-images.githubusercontent.com/2989224/173392443-f76c5c99-07ea-43d7-94fc-d71df5bd796b.png)

Will commit later again on this PR. 

Thanks

**Comment by fuzzybinary** [comment]
Great! Thanks @KarlChow92! Appreciate the help!

**Comment by scopendo** [comment]
> Nevermind, looks like it's:
> 
> ```yaml
> strong-mode: 
>     implicit-dynamic: false
> ```
> 
> I'd love to enable this and fix all the issues at once if possible.



> Nevermind, looks like it's:
> 
> ```yaml
> strong-mode: 
>     implicit-dynamic: false
> ```

Yep, that's the one.  I use the lints from https://pub.dev/packages/very_good_analysis.

**Comment by fuzzybinary** [comment]
Fixed in #148

**Comment by fuzzybinary** [comment]
This is now fixed and deployed in [1.0.0-beta.3](https://pub.dev/packages/datadog_flutter_plugin/versions/1.0.0-beta.3)
